tests/quickchecks.rs

```diff
@@ -104,6 +104,13 @@ fn overlap((a1, a2): (usize, usize), (b1, b2): (usize, usize)) -> bool {
     a1 < b2 && b1 < a2
 }
 
+// Returns whether `(b1, b2)` is contained in `(a1, a2)`.
+fn contains((a1, a2): (usize, usize), (b1, b2): (usize, usize)) -> bool {
+    assert!(a1 < a2);
+    assert!(b1 < b2);
+    a1 <= b1 && a2 >= b2
```

Comment:
nitpick, but I like it this way because you can visually see that `b` is within `a`:

```suggestion
    a1 <= b1 && b2 <= a2
```